app/services/proxy_status.rb

```diff
@@ -0,0 +1,55 @@
+class ProxyStatus
```

Comment:
imho no, it does not own data, rather has a specific function, for example see https://www.netguru.co/blog/service-objects-in-rails-will-help